catalog/transformer/catalog-transformer-microsoft/src/main/java/ddf/catalog/transformer/input/MicrosoftTransformer.java

```diff
@@ -0,0 +1,296 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.input;
+
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.SortedSet;
+
+import javax.imageio.ImageIO;
+import javax.imageio.spi.IIORegistry;
+import javax.xml.transform.Templates;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.tika.io.CloseShieldInputStream;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.mime.MediaType;
+import org.apache.tika.mime.MediaTypeRegistry;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.parser.Parser;
+import org.apache.tika.sax.ToXMLContentHandler;
+import org.codice.ddf.spatial.clavin.ClavinFacade;
+import org.codice.ddf.spatial.clavin.ClavinFacadeException;
+import org.imgscalr.Scalr;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+
+import com.bericotech.clavin.resolver.ResolvedLocation;
+import com.google.common.io.FileBackedOutputStream;
+import com.sun.media.imageioimpl.plugins.jpeg2000.J2KImageReaderSpi;
+import com.sun.media.imageioimpl.plugins.tiff.TIFFImageReaderSpi;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.catalog.transformer.common.tika.MetacardCreator;
+import ddf.catalog.transformer.common.tika.TikaMetadataExtractor;
+
+import net.sf.saxon.TransformerFactoryImpl;
+
+public class MicrosoftTransformer implements InputTransformer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MicrosoftTransformer.class);
+
+    private Templates templates = null;
+
+    private ClavinFacade clavinFacade;
+
+    public MicrosoftTransformer(BundleContext bundleContext, ClavinFacade clavinFacade) {
```

Comment:
Add some javadoc on the ctor. Can the parameters be null? What happens if they are null?